### DISCUSSION TITLE
Is it possible to get historical forecast data from 2021?

### DISCUSSION THREAD

**Discussion by lingqi-su** [open-meteo/open-meteo#353] [Q&A]
Hi, I'm training a machine learning model and uses the weather forecast data as the input. For consistency, it is certainly better to use the histocal weather forecast data instead of actual data. Would it be possible to get my hands on the historical forecast data from 2021 with Open-Meteo? Thanks in advance!

**Comment by patrick-zippenfenig** [comment]
Hi, the Forecast API only keeps the last couple of months. Afterwards data is archived, but the earliest is July 2022.

You can use the [historical weather API](https://open-meteo.com/en/docs/historical-weather-api) with data from 1940 onwards. Unfortunately, the resolution is only 25 km for atmospheric variables like clouds, radiation or wind and 10 km for surface variables like temperature, humidity and soil.

In the next weeks, an additional dataset will follow at 9 km resolution for all weather variables with data from 2017 onwards.

**Reply by farfromrefug** [reply]
@patrick-zippenfenig a question. the historical data which is kept for a few month is the just the forecast computed by the API? or is it the actual weather data gathered after it happened ? trying to see if I can trust the historical data for like rain precipitations. 
thanks

**Reply by patrick-zippenfenig** [reply]
Both. It is a continuous time series from multiple weather model runs. For some models this includes the 3z, 9z, 15z and 21z runs alongside the regular 0z, 6z, 12z and 18z runs. For GFS HRRR it is using every single run, forming a perfect time series of model initialization data which is pretty close to actual measurements. 

Of course you have to account for the gridding bias to 2x2 km boxes (up to 11 km for non Europe and non North America locations). 

With precipitation you found a pain point of this approach as showers can be vert local. Data should be correct for the overall precipitation sum, but the past hour-to-hour data, will be a bit off. In the future I want to
integrate satellite based precipitation datasets like IMERG which should be more reliable.

**Reply by farfromrefug** [reply]
@patrick-zippenfenig thanks for the explanation. dont think I would need past hour by hour. daily would be enough
